pkg/cmd/cli/sa/newtoken.go

```diff
@@ -0,0 +1,260 @@
+account actions by executing them with distinct tokens, to rotate out pre-existing token
+on the service account, or for use by an external client. If a label is provided, it will
+be applied to any creaetd token so that tokens created with this command can be idenitifed.
+`
+
+	newServiceAccountTokenUsage = `%s SA-NAME`
+
+	newServiceAccountTokenExamples = `  # Generate a new token for service account 'default'
+  $ %[1]s 'default'
+
+  # Generate a new token for service account 'default' and apply a label to the new token for identification
+  # %[1]s 'default' --labels 
+`
+)
+
+type NewServiceAccountTokenOptions struct {
+	SAName        string
+	SAClient      unversioned.ServiceAccountsInterface
+	SecretsClient unversioned.SecretsInterface
+
+	Labels map[string]string
+
+	Timeout time.Duration
+
+	Out io.Writer
+	Err io.Writer
+}
+
+func NewCommandNewServiceAccountToken(name, fullname string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &NewServiceAccountTokenOptions{
+		Out:    out,
+		Err:    os.Stderr,
+		Labels: map[string]string{},
+	}
+
+	var requestedLabels string
+	newServiceAccountTokenCommand := &cobra.Command{
+		Use:     fmt.Sprintf(newServiceAccountTokenUsage, name),
+		Short:   newServiceAccountTokenShort,
+		Long:    newServiceAccountTokenLong,
+		Example: fmt.Sprintf(newServiceAccountTokenExamples, fullname),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(args, requestedLabels, f, cmd))
+
+			cmdutil.CheckErr(options.Validate())
+
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+
+	newServiceAccountTokenCommand.Flags().DurationVar(&options.Timeout, "timeout", 30*time.Second, "the maximum time allowed to generate a token")
+	newServiceAccountTokenCommand.Flags().StringVarP(&requestedLabels, "labels", "l", "", "labels to set in all resources for this application, given as a comma-delimited list of key-value pairs")
+	return newServiceAccountTokenCommand
+}
+
+func (o *NewServiceAccountTokenOptions) Complete(args []string, requestedLabels string, f *clientcmd.Factory, cmd *cobra.Command) error {
+	if len(args) != 1 {
+		return cmdutil.UsageError(cmd, fmt.Sprintf("expected one service account name as an argument, got %q", args))
+	}
+
+	o.SAName = args[0]
+
+	if len(requestedLabels) > 0 {
+		labels, err := kubectl.ParseLabels(requestedLabels)
+		if err != nil {
+			return cmdutil.UsageError(cmd, err.Error())
+		}
+		o.Labels = labels
+	}
+
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return fmt.Errorf("could not retrieve default namespace: %v", err)
+	}
+
+	o.SAClient = client.ServiceAccounts(namespace)
+	o.SecretsClient = client.Secrets(namespace)
+	return nil
+}
+
+func (o *NewServiceAccountTokenOptions) Validate() error {
+	if o.SAName == "" {
+		return errors.New("service account name cannot be empty")
+	}
+
+	if o.Labels == nil {
```

Comment:
Because operations on `nil` maps cause a panic and operations on empty ones don't? The entire point of this method is to validate that whatever options you're starting with are workable. I don't want to do a `nil` check in `Run`, as that's the responsibility of `Validate`.